UploadFile/Codeunit/Cod50410.BCPUploadFile.al

```diff
@@ -0,0 +1,57 @@
+codeunit 50410 "BCP UploadFile"
+{
+
+    procedure UploadFile(TableID: Integer; RecordNo: Code[20])
+
+    var
+        uploadTable: Record "BCP Upload";
+        FileInStream: InStream;
+        tempfilename: text;
+        DialogTitleMsg: Label 'Please select a File...';
+
+    begin
+        if UploadIntoStream(DialogTitleMsg, '', 'All Files (*.*)|*.*', tempfilename, FileInStream) then begin
+
+            uploadTable.Init();
+            //clear(uploadtable."BCP File");
+            uploadTable.Insert(true);
+            uploadTable."BCP Table ID" := TableID;
+            uploadTable."BCP No." := RecordNo;
+            uploadTable."BCP File".ImportStream(FileInStream, tempfilename);
+            uploadtable."BCP File Name" := tempfilename;
+            uploadTable.Insert(true);
+            uploadTable.Modify(true);
+            //Message('media uploaded : media id: %1', uploadTable."BCP File".MediaId);
+
+        end;
+    end;
+
+    procedure DownloadFile(tableId: integer; recordNo: code[20]; recid: integer)
+    var
+        tenantMedia: Record "Tenant Media";
+        uploadTable: Record "BCP Upload";
+        ResponseStream: InStream;
+        Filename: text;
+    begin
+        if uploadTable.get(recordNo, tableId, recid) then begin
```

Comment:
It's very hard to read.

```
                else
                    error('No content');
            end
            else
                error('couldn''t find mediaid');
        end
        else
            Error('couldnt get record unfortunately: %1 %2', tableid, recordno);
```

You have several ways to make it better
You can use` if not get then exit` statement
or
just `get()` without `if` will automatically generate error

In both ways you don't need to add end else

Reply:
Alright, i'll make amendment

i actually did it that way when i was troubleshooting to pinpoint why the file was not not downloading.